File: GLAthlete/app/src/main/java/nittcprocon/glathlete/MenuActivity.java
package nittcprocon.glathlete;

import android.content.Intent;
import android.net.Uri;
import android.net.wifi.WifiManager;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.format.Formatter;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.android.gms.appindexing.Action;
import com.google.android.gms.appindexing.AppIndex;
import com.google.android.gms.appindexing.Thing;
import com.google.android.gms.common.api.GoogleApiClient;


public class MenuActivity extends AppCompatActivity {

    private static final String TAG = "MenuActivity";
    private TextView ipAddrTextView;
    private EditText uriEditText;
    private Button startButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        // 初回起動時のみデフォルトの設定値を端末に保存する
        // アプリのエントリポイントにはこれを書いておくこと
        PreferenceManager.setDefaultValues(this, R.xml.preferences, false);

        uriEditText = (EditText) findViewById(R.id.uriEditText);
        startButton = (Button) findViewById(R.id.startButton);
        ipAddrTextView = (TextView) findViewById(R.id.ipAddrTextView);

        ipAddrTextView.setText(getLocalIpAddress());
        startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startVRActivity(uriEditText.getText().toString());
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem) {
        switch (menuItem.getItemId()) {
            case R.id.action_settings:
                Intent intent = new Intent(this, SettingsActivity.class);
                startActivity(intent);
                return true;

            default:
                return super.onOptionsItemSelected(menuItem);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        return true;
    }

    private void startVRActivity(String uri) {
        Intent intent = new Intent(this, VRActivity.class);
        intent.putExtra("uri", uri);
        startActivity(intent);
    }

    // http://stackoverflow.com/questions/6064510/how-to-get-ip-address-of-the-device
    // FIXME: Wi-Fiにしか対応してない
    // FIXME: deprecatedなメソッドを使ってる
    private String getLocalIpAddress() {
        WifiManager wm = (WifiManager) getSystemService(WIFI_SERVICE);
        return Formatter.formatIpAddress(wm.getConnectionInfo().getIpAddress());
    }
}

File: GLAthlete/app/src/main/java/nittcprocon/glathlete/MenuActivity.kt
package nittcprocon.glathlete

import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.text.format.Formatter
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MenuActivity : AppCompatActivity() {
    private val TAG = "MenuActivity"
    private var ipAddrTextView: TextView? = null
    private var uriEditText: EditText? = null
    private var startButton: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        // 初回起動時のみデフォルトの設定値を端末に保存する
        // アプリのエントリポイントにはこれを書いておくこと
        PreferenceManager.setDefaultValues(this, R.xml.preferences, false)

        MyContexts.applicationContext = applicationContext

        uriEditText = findViewById(R.id.uriEditText) as EditText
        startButton = findViewById(R.id.startButton) as Button
        ipAddrTextView = findViewById(R.id.ipAddrTextView) as TextView

        ipAddrTextView!!.text = localIpAddress
        startButton!!.setOnClickListener { startVRActivity(uriEditText!!.text.toString()) }
    }

    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
        when (menuItem.itemId) {
            R.id.action_settings -> {
                val intent = Intent(this, SettingsActivity::class.java)
                startActivity(intent)
                return true
            }

            else -> return super.onOptionsItemSelected(menuItem)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    private fun startVRActivity(uri: String) {
        val intent = Intent(this, VRActivity::class.java)
        intent.putExtra("uri", uri)
        startActivity(intent)
    }

    // http://stackoverflow.com/questions/6064510/how-to-get-ip-address-of-the-device
    // FIXME: Wi-Fiにしか対応してない
    // FIXME: deprecatedなメソッドを使ってる
    private val localIpAddress: String
        get() {
            val wm = getSystemService(Context.WIFI_SERVICE) as WifiManager
            return Formatter.formatIpAddress(wm.connectionInfo.ipAddress)
        }
}
